chain/network/src/peer.rs

```diff
@@ -605,10 +611,36 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for Peer {
         let msg_size = msg.len();
 
         self.tracker.increment_received(msg.len() as u64);
+
         let mut peer_msg = match bytes_to_peer_message(&msg) {
             Ok(peer_msg) => peer_msg,
             Err(err) => {
-                info!(target: "network", "Received invalid data {:?} from {}: {}", msg, self.peer_info, err);
+                if let Some(version) = err
+                    .get_ref()
+                    .map(|err| err.downcast_ref::<HandshakeFailureReason>())
+                    .flatten()
```

Comment:
maybe it's makes sense to use `flat_map`?

Reply:
done, it is called `and_then` in rust, just discovered them :)